src/org/traccar/model/Position.java

```diff
@@ -22,9 +22,11 @@
     public static final String KEY_ORIGINAL = "raw";
     public static final String KEY_INDEX = "index";
     public static final String KEY_HDOP = "hdop";
-    public static final String KEY_SATELLITES = "sat";
+    public static final String KEY_SATELLITES = "sat"; // in use
+    public static final String KEY_SATELLITES_VISIBLE = "satVisible";
     public static final String KEY_RSSI = "rssi";
     public static final String KEY_GPS = "gps";
+    public static final String KEY_FIX = "fix"; // GPS, GSM, none
```

Comment:
The EV-07S device can provide location information (aka "a fix") either by GPS or by GSM triangulation.  Currently, the protocol will flag location information not provided by GPS as invalid.  I have maintained this behavior for now as I am testing the accuracy of the "GSM Locating" capability.  If the accuracy is decent, I will propose marking these locations as valid.